### PR TITLE
Feature/handlebars

### PR DESCRIPTION
**Pull request by marcinkp**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,6 +18,7 @@ fork.properties
 *.jar
 *.war
 *.ear
+!*/wiremock/libs/handlebars-4.1.2.jar
```
Comment (pun-ky) [inline .gitignore:21]: !wiremock/libs/\*\*/\*.jar

File: wiremock/build.gradle.kts [hunk 2]
```diff
@@ -67,7 +67,7 @@ tasks {
                     "com.jayway.jsonpath.spi.mapper"
             )
             embedPackage("net.sf.jopt-simple:jopt-simple:5.0.4", "joptsimple.*")
-            embedPackage("com.github.jknack:handlebars:4.1.2",
+            embedPackage(files("libs/handlebars-4.1.2.jar"),
```
Comment (pun-ky) [inline wiremock/build.gradle.kts:70]: nice one :)

File: wiremock/src/integTest/java/com/cognifide/aem/stubs/wiremock/HelloWorldTest.java [hunk 3]
```diff
@@ -61,12 +61,22 @@ public void shouldGetFineWithBody() {
   }
 
   @Test
-  public void shouldGetFineWithBodyTemplated() {
+  public void shouldGetFineWithBodyTemplatedByPebble() {
```
Comment (pun-ky) [inline wiremock/src/integTest/java/com/cognifide/aem/stubs/wiremock/HelloWorldTest.java:64]: pebble should stay only if admin mappings listing / serialization will work / not fail

Comment (pun-ky) [inline wiremock/src/integTest/java/com/cognifide/aem/stubs/wiremock/HelloWorldTest.java:64]: pebble should stay only if admin mappings listing / serialization will work / not fail

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockOptionsFactory.java [hunk 4]
```diff
@@ -32,11 +36,11 @@ private String getRootPath() {
...
-  private void addExtensions() {
+  private List<Extension> transformers() {
     JcrFileReader jcrFileReader = new JcrFileReader(stubs.getResolverAccessor(), getRootPath());
-    extensions.putAll(ExtensionLoader.asMap(
-      Collections.singletonList(new PebbleTransformer(jcrFileReader,
-        stubs.getConfig().globalTransformer())))
-    );
+    return ImmutableList.<Extension>builder()
+      .add(new PebbleTransformer(jcrFileReader, stubs.getConfig().globalTransformer() == PEBBLE))
+      .add(new ResponseTemplateTransformer(stubs.getConfig().globalTransformer() == HANDLEBARS))
```
Comment (pun-ky) [inline wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockOptionsFactory.java:43]: handlebars global by default, nice

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockStubs.java [hunk 5]
```diff
@@ -206,10 +208,10 @@ public Config getConfig() {
 
     @AttributeDefinition(
       name = "Global Template Transformer",
-      description = "Enables Pebble template engine / templating"
+      description = "Enables template engine / templating. Handlebars and Pebbles engines are supported"
         + " for response body content and file paths when loading body files. Effectively enables dynamic file loading"
         + " instead of preloading and simplifies defining stubs.")
-    boolean globalTransformer() default true;
+    TransformerEngine globalTransformer() default HANDLEBARS;
```
Comment (pun-ky) [inline wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockStubs.java:214]: hmm you want to have a nice switch between these two engines, ok.

**Review by pun-ky — APPROVED** [review]

**Review by pun-ky — APPROVED** [review]
I've been trying with sth similar. looks nice, cool ;)

**Comment by pun-ky** [comment]
CI fails? 

```

* What went wrong:
Execution failed for task ':moco:pmdMain'.
> java.lang.NoClassDefFoundError: io/netty/handler/ssl/SslHandler
```

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockOptionsFactory.java [hunk 6]
```diff
@@ -1,14 +1,18 @@
 package com.cognifide.aem.stubs.wiremock;
 
+import static com.cognifide.aem.stubs.wiremock.TransformerEngine.HANDLEBARS;
+import static com.cognifide.aem.stubs.wiremock.TransformerEngine.PEBBLE;
```
Comment (pun-ky) [inline wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockOptionsFactory.java:4]: too detailed import / unusual ;)

File: wiremock/src/integTest/java/com/cognifide/aem/stubs/wiremock/HelloWorldTest.java (hunk 3 above)
Comment (marcinkp) [reply wiremock/src/integTest/java/com/cognifide/aem/stubs/wiremock/HelloWorldTest.java:64]: solved. Introduce DynamicParameterProvider

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockStubs.java (hunk 5 above)
Comment (marcinkp) [reply wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockStubs.java]: yes. always better then string ;)

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockOptionsFactory.java (hunk 6 above)
Comment (marcinkp) [reply wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockOptionsFactory.java]: it depends. I reorganized it a bit

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/transformers/DynamicParameterProvider.java [hunk 7]
```diff
@@ -48,6 +48,11 @@ private ClousureCall(Closure closure) {
     public Object call() {
       return closure.call();
     }
+
+    @Override
+    public String toString(){
+      return Closure.class.getName();
```
Comment (pun-ky) [inline wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/transformers/DynamicParameterProvider.java:54]: Are you sure here? Above you just replaced closure to dynamic provider

Comment (marcinkp) [reply wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/transformers/DynamicParameterProvider.java:54]: as we support groovy Closure and java Supplier

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockStubs.java [hunk 8]
```diff
@@ -206,10 +178,10 @@ public Config getConfig() {
 
     @AttributeDefinition(
       name = "Global Template Transformer",
-      description = "Enables Pebble template engine / templating"
+      description = "Enables template engine / templating. Handlebars and Pebbles engines are supported"
```
Comment (pun-ky) [inline wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/WireMockStubs.java:181]: Pebble! not pebbles :D

File: wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/transformers/DynamicParameterProvider.java (hunk 7 above)
Comment (pun-ky) [reply wiremock/src/main/java/com/cognifide/aem/stubs/wiremock/transformers/DynamicParameterProvider.java:54]: I mean, toStriing impl

**Pull request merged**